sda/processor.go

```diff
@@ -25,13 +116,16 @@ type ServiceProcessor struct {
 
 // NewServiceProcessor initializes your ServiceProcessor.
 func NewServiceProcessor(c *Context) *ServiceProcessor {
-	return &ServiceProcessor{
+	s := &ServiceProcessor{
```

Comment:
Is this for debugging purposes? Then add a `log.Lvl3(s)`, else I'd put the return.